base_layer/wallet_ffi/src/lib.rs

```diff
@@ -860,6 +861,49 @@ pub unsafe extern "C" fn seed_words_create() -> *mut TariSeedWords {
     Box::into_raw(Box::new(TariSeedWords(Vec::new())))
 }
 
+/// Create a TariSeedWords instance containing the entire mnemonic wordlist for the requested language
+///
+/// ## Arguments
+/// `language` - The pointer to a TariMnemonicLanguage
+/// `error_out` - Pointer to an int which will be modified to an error code should one occur, may not be null. Functions
+/// as an out parameter.
+///
+/// ## Returns
+/// `TariSeedWords` - Returns the TariSeedWords instance containing the entire mnemonic wordlist for the requested
+/// language.
+///
+/// # Safety
+/// The ```string_destroy``` method must be called when finished with a string from rust to prevent a memory leak
+#[no_mangle]
+pub unsafe extern "C" fn get_mnemonic_wordlist_for_language(
+    language: *const TariMnemonicLanguage,
```

Comment:
The problem with this is that how does the FFI client construct this enum? I think the best is to accept a c string here and parse it to the enum within the function. If the string is null or doesn't pass then you will return an error across the FFI boundary.

Reply:
Agreed